qiskit/utils/deprecation.py

```diff
@@ -434,3 +443,29 @@ def add_deprecation_to_docstring(
     else:
         original_lines.extend(new_lines)
     func.__doc__ = "\n".join(original_lines)
+
+
+def remove_sphinx_from_runtime_deprecation_warning(msg: str) -> str:
+    """Remove Sphinx-isms from ``msg`` so that it's more readable in runtime warnings.
+
+    This function is useful so that you can write deprecation messages using Sphinx syntax like
+    ``:class:`` references, so that the deprecation renders more nicely in docs. But, that syntax
+    is not very readable when we show the same message as a runtime warning, so this function
+    allows you to strip it before calling ``warnings.warn()``.
+
+    This function will:
+
+    * Replace `` with `
+    * Remove roles like `:ref:`, `:class:`, and `:py:func:`.
+
+    Feel free to continue improving this function to do more normalization.
+
+    Args:
+        msg: The deprecation message.
+
+    Returns:
+        The original ``msg`, but normalized.
+    """
+    msg = msg.replace("``", "`")
+    # Match both `:role:` and `:py:role:`.
+    return re.sub(r":[a-z]+:([a-z]+:)?", "", msg)
```

Comment:
I think both the domain and the role identifiers can be much more free-form than ASCII lowercase, but I don't know the exact rules.  Certainly intersphinx uses the `+` character in the role position (e.g. `` :external:retworkx+ref:`PyDiGraph` ``), and the key is defined by the Sphinx conf value `intersphinx_mapping`, so in Python terms it can be any Unicode though I assume Sphinx places _some_ additional restrictions on it.